src/services/organizeImports.ts

```diff
@@ -47,8 +55,73 @@ namespace ts.OrganizeImports {
         return changeTracker.getChanges();
     }
 
-    function removeUnusedImports(oldImports: ReadonlyArray<ImportDeclaration>) {
-        return oldImports; // TODO (https://github.com/Microsoft/TypeScript/issues/10020)
+    function removeUnusedImports(oldImports: ReadonlyArray<ImportDeclaration>, sourceFile: SourceFile, program: Program) {
+        const typeChecker = program.getTypeChecker();
+        const jsxNamespace = typeChecker.getJsxNamespace();
+        const jsxContext = sourceFile.languageVariant === LanguageVariant.JSX && program.getCompilerOptions().jsx;
+
+        const usedImports: ImportDeclaration[] = [];
+
+        for (const importDecl of oldImports) {

```

Comment:
`mapDefined(usedImports, importDecl => <<<importDecl, updated import, or undefined>>>)`

Reply:
I'm not sure I understand the advantage of doing it that way.